### PR TITLE
feat!: computed members and audit log config added

### PR DESCRIPTION
**Pull request by waxb**
- chore: update module from template and tests
- feat!: computed members and audit log config added
- chore: update changelog

### CODE REVIEW

File: test/terramate_google.tm.hcl [hunk 1]
```diff
@@ -0,0 +1,78 @@
+globals {
+  minimum_provider_version    = "4.0"
+  provider_version_constraint = "~> ${global.minimum_provider_version}"
+
```
Comment (mariux) [inline test/terramate_google.tm.hcl:4]: Let's also right away drop support for provider < 4.0 and terraform < 1.0 in versions.tf

**Review by mariux — CHANGES_REQUESTED** [review]
- policy is missing support for computed members.
- docs need to be updated to reflect audit logs are not bound to audit_log resource. (see folder-iam)
- drop support for last gen provider and terraform versions

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -16,7 +17,121 @@ jobs:
...
+        # actions/setup-go@v3.2.0 -> b22fbbc2921299758641fab08929b4ac52b32923
+        uses: actions/setup-go@b22fbbc2921299758641fab08929b4ac52b32923
+        with:
+          go-version: 1.17.6
+
+      - name: Run Unit Tests
+        if: steps.changes.outputs.terraform == 'true'
+        run: make test/unit-tests
+        env:
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.MINEIROS_TESTING_AWS_SECRET_ACCESS_KEY }}
```
Comment (zied-elouaer) [inline .github/workflows/tests.yml:132]: Remove AWS?

File: README.md [hunk 3]
```diff
@@ -86,7 +86,7 @@ See [variables.tf] and [examples/] for details and use-cases.
 
 #### Main Resource Configuration
 
-- [**`org_id`**](#var-org_id): *(**Required** `string`)*<a name="var-org_id"></a>
+- [**`org_id`**](#var-org_id): *(Optional `string`)*<a name="var-org_id"></a>
 
   The organization ID. If not specified, terraform uses the ID of the organization configured with the provider.
```
Comment (zied-elouaer) [inline README.md:91]: Didn't we agree that the org_id is NOT an attribute of the provider config?

File: .github/workflows/tests.yml (hunk 2 above)
Comment (waxb) [reply .github/workflows/tests.yml:132]: was thinking about it but it resolves to `empty` anyway. will bring it up next time

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -77,14 +77,19 @@ If everything looks good, they will merge the code and release a new version whi
 
 <!-- References -->
 
-[pull requests]: https://github.com/mineiros-io/terraform-google-organization-iam/pulls
-[pre-commit-file]: https://github.com/mineiros-io/terraform-google-organization-iam/blob/main/.pre-commit-config.yaml
+<!-- markdown-link-check-disable -->
```
Comment (soerenmartius) [inline CONTRIBUTING.md:80]: This isn't the latest version of the file. You may want to update the module locally :)

Comment (waxb) [reply CONTRIBUTING.md]: forgot to commit! done

File: README.tfdoc.hcl [hunk 5]
```diff
@@ -128,20 +128,33 @@ section {
...
+          #     "user:member@example.com",
+          #     "computed:myserviceaccount",
+          #   ]
+          #   computed_members_map = {
+          #     myserviceaccount = "serviceAccount:${google_service_account.service_account.id}"
+          #   }
+          # END
         }
 
-        variable "project" {
```
Comment (mariux) [inline README.tfdoc.hcl:141]: nice catch :) haha

File: variables.tf [hunk 6]
```diff
@@ -25,8 +25,19 @@ variable "members" {
...
+  }
+}
+
+variable "computed_members_map" {
+  type        = map(string)
+  description = "(Optional) A map of members to replace in 'var.members' or in members of 'var.policy_bindings' to handle terraform computed values."
+  default     = {}
+
+  validation {
+    condition     = alltrue([for k, v in var.computed_members_map : can(regex("^(allUsers|allAuthenticatedUsers|(user|serviceAccount|group|domain|computed):)", v))])
```
Comment (mariux) [inline variables.tf:39]:
```suggestion
    condition     = alltrue([for k, v in var.computed_members_map : can(regex("^(allUsers|allAuthenticatedUsers|(user|serviceAccount|group|domain):)", v))])
```

File: variables.tf [hunk 7]
```diff
@@ -25,8 +25,19 @@ variable "members" {
...
+    error_message = "The value must be a non-empty string being a valid principal type identified with `allUsers`, `allAuthenticatedUsers` or prefixed with `user:`, `serviceAccount:`, `group:`, `domain:` or `computed:`."
```
Comment (mariux) [inline variables.tf:40]:
```suggestion
    error_message = "The value must be a non-empty string being a valid principal type identified with `allUsers`, `allAuthenticatedUsers` or prefixed with `user:`, `serviceAccount:`, `group:`, or `domain:`."
```

**Review by mariux — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml [hunk 8]
```diff
@@ -16,7 +17,121 @@ jobs:
...
+        run: go install github.com/mineiros-io/terradoc/cmd/terradoc@af1a7b3ae3635958adf5ee2f40e0c3e70fd0803a
+        env:
+          GOPROXY: direct
+
+      - name: Set up Terraform
+        # for security reasons we pin commit ids and not tags.
+        # hashicorp/setup-terraform@v2.0.0 -> 17d4c9b8043b238f6f35641cdd8433da1e6f3867
+        uses: hashicorp/setup-terraform@17d4c9b8043b238f6f35641cdd8433da1e6f3867
+        with:
+          terraform_version: 1.2.1
```
Comment (zied-elouaer) [inline .github/workflows/tests.yml:61]: Why are we using an older version here?

**Review by zied-elouaer — APPROVED** [review]

**Review by mariux — CHANGES_REQUESTED** [review]
changelog is missing the breaking changes

File: .github/workflows/tests.yml (hunk 8 above)
Comment (waxb) [reply .github/workflows/tests.yml:61]: https://github.com/mineiros-io/terraform-module-template/blob/553f36cb66dbebf343e8ad3217883c3a40a1d53a/.github/workflows/tests.yml#L61

we can update the module template

Comment (zied-elouaer) [reply .github/workflows/tests.yml:61]: I think we should. The tests we have should be enough to discover any regressions related to the TF version.

**Review by mariux — APPROVED** [review]
lgtm

File: .github/workflows/tests.yml (hunk 8 above)
Comment (mariux) [reply .github/workflows/tests.yml:61]: we do not use any newer features and just require >= 1.0 ... so any terraform version should work.. ;)

**Pull request merged**